ballerina/caller.bal

```diff
@@ -22,15 +22,23 @@ import ballerina/jballerina.java;
 # + remotePort - The port number of the remote host
 # + localHost - The bound hostname
 # + localPort - The port number to which the socket is bound
-public client class Caller {
+# + id - The unique ID associated with the connection
+public isolated client class Caller {
 
-  public string remoteHost = "";
-  public int remotePort = -1;
-  public string localHost = "";
-  public int localPort = -1;
+  public final string remoteHost;
+  public final int remotePort;
+  public final string localHost;
+  public final int localPort;
+  public final string id;
```

Comment:
Why are we exposing these config as public constants. Isn't it preferred to make these private and have getters ?

Reply:
The designed API is `caller.id`, `caller. remotePort` etc. And these are not configs.